crates/sui-node/src/metrics.rs

```diff
@@ -136,6 +152,9 @@ pub fn start_metrics_push_task(config: &sui_config::NodeConfig, registry: Regist
             interval.tick().await;
 
             if let Err(error) = push_metrics(&client, &url, &registry).await {
+                // aggressively recreate our client connection if we hit an error
+                // since our tick interval is only every min, this should not be racey
+                client = client.recreate();
```

Comment:
@bmwill i tried your suggestion about just constructing a new type, but there were borrow problems, i think because of the move block here.  It seems easier to keep the recreate call i added and just add a mut.  what do you think?  I'm happy to still try your last suggestion:

> client = MetricsPushClient::new(config.network_key_pair().copy());

I just couldn't get it to compile w/o errors